StyleCop.Analyzers/StyleCop.Analyzers.Test/DocumentationRules/SA1655UnitTests.cs

```diff
@@ -0,0 +1,356 @@
+        {
+            var testCode = @"
+/// <summary>
+/// <code>Code.</code>
+/// Summary.
+/// </summary>
+public class ClassName
+{
+}";
+
+            // reported as SA1654
+            await this.VerifyCSharpDiagnosticAsync(testCode, EmptyDiagnosticResults, CancellationToken.None).ConfigureAwait(false);
+        }
+
+        [Fact]
+        public async Task TestCodeAndInlineParagraphInRemarksAsync()
+        {
+            var testCode = @"
+/// <remarks>
+/// <code>Code.</code>
+/// Remarks.
+/// </remarks>
+public class ClassName
+{
+}";
+
+            // reported as SA1653
+            await this.VerifyCSharpDiagnosticAsync(testCode, EmptyDiagnosticResults, CancellationToken.None).ConfigureAwait(false);
+        }
+
+        [Fact]
+        public async Task TestThreeInlineParagraphsWithOtherElementsAsync()
+        {
+            var testCode = @"
+/// <summary>
+/// Leading summary.
+/// <code>Code.</code>
+/// <para>Summary.</para>
+/// <note>Note.</note>
+/// Closing summary.
+/// </summary>
+public class ClassName
+{
+}";
+
+            // reported as SA1653 and SA1654
+            await this.VerifyCSharpDiagnosticAsync(testCode, EmptyDiagnosticResults, CancellationToken.None).ConfigureAwait(false);
+        }
+
+        [Fact]
+        public async Task TestThreeInlineParagraphsWithOtherElementsInRemarksAsync()
+        {
+            var testCode = @"
+/// <remarks>
+/// Leading remarks.
+/// <code>Code.</code>
+/// <para>Remarks.</para>
+/// <note>Note.</note>
+/// Closing remarks.
+/// </remarks>
+public class ClassName
+{
+}";
+
+            // reported as SA1653
+            await this.VerifyCSharpDiagnosticAsync(testCode, EmptyDiagnosticResults, CancellationToken.None).ConfigureAwait(false);
+        }
+
+        [Fact]
+        public async Task TestSeeIsAnInlineElementAsync()
+        {
+            var testCode = @"
+/// <summary>
+/// Leading summary.
+/// <see cref=""ClassName""/>.
+/// </summary>
+public class ClassName
+{
+}";
+
+            await this.VerifyCSharpDiagnosticAsync(testCode, EmptyDiagnosticResults, CancellationToken.None).ConfigureAwait(false);
+        }
+
+        [Fact]
+        public async Task TestExceptionElementsAsync()
+        {
+            var testCode = @"
+using System;
+public class ClassName
+{
+    /// <exception cref=""ArgumentNullException"">If <paramref name=""x""/> is <see langword=""null""/>.</exception>
+    /// <exception cref=""ArgumentException"">
+    /// If <paramref name=""x""/> is empty.
+    /// <para>-or-</para>
+    /// <para>If <paramref name=""y""/> is empty.</para>
+    /// </exception>
+    public void MethodName(string x, string y)
+    {
+    }
+}";
+
+            var fixedCode = @"
+using System;
+public class ClassName
+{
+    /// <exception cref=""ArgumentNullException""><para>If <paramref name=""x""/> is <see langword=""null""/>.</para></exception>
+    /// <exception cref=""ArgumentException"">
+    /// If <paramref name=""x""/> is empty.
```

Comment:
Yes, as part of SA1654 it would be wrapped in `<para>` elements, but this test is for SA1655. SA1655 says the documentation for `ArgumentException` contains blocks, and therefore the documentation for all `<exception>` elements should be written as blocks. The diagnostic is reported on `ArgumentNullException` since it contains *only* inline content.